doc/doxygen/snippets/meshhierarchy.cc

```diff
@@ -0,0 +1,42 @@
+/**
+ * @file
+ * @brief Doxygen snippets to show lf::assemble::COOMatrix usage
+ * @author Ralf Hiptmair
+ * @date Tue 29 Oct 2019 03:09:10 PM CET
+ * @copyright MIT License
+ */
+
+#include <lf/mesh/test_utils/test_meshes.h>
+#include <lf/refinement/refinement.h>
+
+namespace lf::refinement {
+void snippetmh(void) {
+  //! [usage]
+  // Generate test mesh
+  std::shared_ptr<lf::mesh::Mesh> mesh_p =
+      lf::mesh::test_utils::GenerateHybrid2DTestMesh(0);
+  // Construction of a mesh hierarchy requires a factory object
+  std::unique_ptr<lf::mesh::hybrid2d::MeshFactory> mesh_factory_ptr =
+      std::make_unique<lf::mesh::hybrid2d::MeshFactory>(2);
+  // Initialize still flat MESH HIERARCHY containing a single mesh
+  lf::refinement::MeshHierarchy multi_mesh(mesh_p, std::move(mesh_factory_ptr));
+  // Perform a first step of regular refinement: adds a mesh
+  multi_mesh.RefineRegular();
+  // For demonstration purposes: Mark edges whose center lies inside a square
+  std::function<bool(const lf::mesh::Mesh &, const lf::mesh::Entity &edge)>
```

Comment:
Just a reminder that from a performance point of view this is suboptimal. std::function brings a lot of overhead with it. This is one of the few cases where `auto` cannot be replaced with an explicit type (as far as I know).

Reply:
Thanks for pointing this out: https://stackoverflow.com/questions/14677997/stdfunction-vs-template.
I had not idea of the performance penalty. Changed now.